lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -609,13 +609,99 @@ class GetDimensionSizeToConstantConversionPattern
   }
 };
 
+struct SelectToSliceConversionPattern
```

Comment:
Can you give example above how does this decomp look like given some input?

Reply:
Yes, done.